qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -437,46 +440,51 @@ def add_analysis_callback(self, callback: Callable, **kwargs: Any):
             self._analysis_callbacks[cid] = AnalysisCallback(
                 name=callback.__name__,
                 callback_id=cid,
-                status=AnalysisStatus.QUEUED,
             )
 
-            # Get futures to wait for before running analysis
-            if self._analysis_futures:
-                # If analysis futures are present wait for all previous analysis
-                # futures to complete before running the latest one
-                futs = self._analysis_futures.values()
-            else:
-                # Create a dummy future to wait for jobs to finish.
-                # This allows subsequent analysis futures to be cancelled as they
-                # will be queued by the ThreadPoolExecutor while this future
-                # is still running
-                futs = [self._analysis_executor.submit(self._wait_for_jobs)]
+            # Futures to wait for
+            futs = self._job_futures.values() + self._analysis_futures.values()
+            wait_future = self._monitor_executor.submit(
+                self._wait_for_futures, futs, name="jobs and analysis"
+            )
+
+            # Create a future to monitor event for calls to cancel_analysis
+            def _monitor_cancel():
+                self._analysis_callbacks[cid].event.wait()
```

Comment:
There is no expectation that this future should ever timeout. It should only terminate if cancel_analysis is called, or jobs finish and the callback starts (both of which call `event.set()` causing this wait to end).

It would only make sense to add if a timeout kwarg was added to the `add_analysis_callback` method.